### PR TITLE
add class CurrencyManager: ьÑconsctructor, method getAllCurrency, cal…

### PR DESCRIPTION
**Pull request by katerinaonix8585**
…culateRate, calculateTotalAmount

### CODE REVIEW

File: src/BL/CurrecytManager.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
+package BL;
+
+import java.util.HashMap;
+
+public class CurrencyManager {
+    //Определяет статическую константу - значение валюты по умолчанию
+    public static final CurrencyTitle defaultCurrency = CurrencyTitle.EUR;
```
Comment (ychepel) [inline src/BL/CurrecytManager.java]: this variable is a constant and it should be written in UPPER_SNAKE_CASE

File: src/BL/CurrecytManager.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    // Определяет приватное поле data, предназначенное для хранения экземпляра класса CurrencyExchangeHandler
+    private final CurrencyExchangeHandler data;
+
+    // Конструктор
+    public CurrentManager(String rateFilePath) {
+        this.data = new CurrencyExchangeHandler(rateFilePath);
```
Comment (ychepel) [inline src/BL/CurrecytManager.java]: here we should not take the file path as an argument. at the business layer, a class can store an address. and it’s better to put this address in a new constant above

File: src/BL/CurrecytManager.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    /**
+     * Метод getAllCurrencies - возвращает все валюты и курсы их обмена по отношению к дефолтной валюте.
+     * Использует метод read() класса CurrencyExchangeHandler.
+     * Метод возвращает HashMap с валютами и их курсами.
+    **/
+     public HashMap<String, Double> getAllCurrency () {
```
Comment (ychepel) [inline src/BL/CurrecytManager.java]: the key of HashMap should be CurrencyTitle type

File: src/BL/CurrecytManager.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        HashMap<CurrencyTitle, Double> exchangeRate = data.read();
+
+        //TODO - предусмотреть случай, когда курс = 0
+
+        if (initialCurrency.equals(defaultCurrency)) {
+            return 1 / exchangeRate.get(resultCurrency);
+        } else if (resultCurrency.equals(defaultCurrency)) {
+            return exchangeRate.get(initialCurrency);
+        } else {
+            return exchangeRate.get(initialCurrency) / exchangeRate.get(resultCurrency);
```
Comment (ychepel) [inline src/BL/CurrecytManager.java]: here, isn't it the other way around to divide? :)

**Review by ychepel — COMMENTED** [review]
please check my comments

File: src/BL/CurrecytManager.java (hunk 1 above)
Comment (katerinaonix8585) [reply src/BL/CurrecytManager.java]: done

File: src/BL/CurrecytManager.java (hunk 4 above)
Comment (katerinaonix8585) [reply src/BL/CurrecytManager.java]: Here is my calculation. Perhaps I have a logical error somewhere...
Calculation

File: src/BL/CurrecytManager.java (hunk 2 above)
Comment (katerinaonix8585) [reply src/BL/CurrecytManager.java]: done

File: src/BL/CurrecytManager.java (hunk 3 above)
Comment (katerinaonix8585) [reply src/BL/CurrecytManager.java]: done

File: src/BL/CurrencyManager.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+import DL.CurrencyExchangeHandler;
+import DL.CurrencyTitle;
+import java.util.HashMap;
+
+public class CurrencyManager {
+    //Определяет статическую константу - значение валюты по умолчанию
+    public static final CurrencyTitle DEFAULT_CURRENCY = CurrencyTitle.EUR;
+    // Определяет приватное поле data, предназначенное для хранения экземпляра класса CurrencyExchangeHandler
+    private String rateFilePath = "src/rate.txt";
```
Comment (ychepel) [inline src/BL/CurrencyManager.java]: this is the constant  (private static final + upper snake case in name)

Comment (ychepel) [reply src/BL/CurrencyManager.java]: and the name of our file is rates.txt :)

File: src/BL/CurrencyManager.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    private final CurrencyExchangeHandler data;
+
+    // Конструктор
```
Comment (ychepel) [inline src/BL/CurrencyManager.java]: comments like this is obvious and no need to add them

File: src/BL/CurrencyManager.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+     public HashMap<CurrencyTitle, Double> getAllCurrency () {
+        return data.read();
+    }
+
+    /** Метод calculateRate - рассчитывает курс обмена между двумя валютами
+     * Параметры метода: initialCurrency начальная валюта, resultCurrency  конечная валюта
+     * Метод возвращает переменную типа double - курс обмена между заданными валютами
+     **/
+    public double calculateRate (CurrencyTitle initialCurrency, CurrencyTitle resultCurrency){
+        HashMap<CurrencyTitle, Double> exchangeRate = data.read();
```
Comment (ychepel) [inline src/BL/CurrencyManager.java]: it is better to use the plural form for collection variable (exchangeRates)

**Review by ychepel — COMMENTED** [review]
please fix these tiny issues

**Comment by katerinaonix8585** [comment]
@ychepel. All corrections done. After changes in the calculateRate method, the question arises: do we still need the constant DEFAULT_CURRENCY?

**Comment by ychepel** [comment]
I also looked at this default value and have not yet understood why we need it :)
You can probably delete it for now.

**Review by ychepel — APPROVED** [review]

**Pull request merged**